### PR TITLE
Feature 963 quick search

### LINKED ISSUE #963: Fix documentation quick search links
The quick search page (docs/Users_Guide/quicksearch.rst and use case documentation files (docs/use_cases) have hard-coded URLs to perform the search. They point to the develop version of the docs. These have to be updated when a new release is created. This also prevents testing that links work properly in a branch because the changed content doesn't exist in develop yet.

**Some of the URLs in the use case doc files still use dtcenter.github.io URLs which are broken!**

## Describe the Enhancement ##
Some thought/investigation is needed to come up with a good solution. My initial thoughts:

* URLs can be relative instead of a full hard-coded path
* URLs that start with / are relative to the "root" of the webpage. We should investigate what the "root" of a ReadTheDocs page is. Typically the root is the top level, i.e. metplus.readthedocs.io but we need to know the rest of the URL for the current branch/tag/version, i.e. metplus.readthedocs.io/en/develop. If ReadTheDocs somehow sets the root to be this location, then we can easily create paths relative to that for the searches. I think this would be the ideal solution if that is the case.
* URLs that start with ../ are relative to the current page. This would work great for the quicksearch.rst page, however the use case doc pages will have a different relative path. We typically copy links from the quicksearch page and paste them into the use case doc files (and modify the link text to match the actual keyword) so this would involve an extra step to add extra "../" to get the correct relative path. One option is to remove the links from the use case doc pages since the actual keyword text is the only required text so that the search finds the page. The URL on those pages are just for convenience to search for other related cases. We could add a link to the quick search page so that users can easily get to that page to search for that keyword.

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by lisagoodrich**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Keywords have been added into a note box.
Keywords have been bulleted for easier reading.
The broken links have been removed.  
All links have been removed except link to "quick search" section. Below is the new language and link.
"Navigate to the METplus[ Quick Search](https://metplus.readthedocs.io/en/feature_963_quick_search/Users_Guide/quicksearch.html#quick-search) for Use Cases page to discover other similar use cases."

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Please double check all
Keywords are in a blue note box
All Keywords have a bullet
The bottom of the note has the text: "Navigate to the METplus[ Quick Search](https://metplus.readthedocs.io/en/feature_963_quick_search/Users_Guide/quicksearch.html#quick-search) for Use Cases page to discover other similar use cases."
Per Julie, she found a typo on one of the pages so that will need approval too.

PLEASE NOTE: All "sphinx_gallery_thumbnail_path =...." have been moved to below the Keyword notes for consistency.   It might look better to hide this in a future version.  We found this documentation but were't able to get it to work. [Hiding sphinx-gallery_thumbnail_path](https://sphinx-gallery.github.io/stable/configuration.html)

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes or No]**.  Yes.  There were no warnings or errors.

- [x] Do these changes include sufficient testing updates? **[Yes or No]** Yes

- [x] Will this PR result in changes to the test suite? **[Yes or No]**</br> No
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **7/16/2121**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/use_cases/model_applications/convection_allowing_models/MODE_fcstHRRR_obsMRMS_Hail_GRIB2.py [hunk 1]
```diff
@@ -134,12 +134,19 @@
...
 #
+# .. note::
+#
+#   * MODEToolUseCase 
+#   * ConvectionAllowingModelsAppUseCase
+#   * GRIB2FileUseCase 
+#   * RegriddingInToolUseCase 
+#   * NOAAHWTOrgUseCase 
+#   * NCAROrgUseCase 
+#   * nDiagnosticsUseCase
```
Comment (jprestop) [inline docs/use_cases/model_applications/convection_allowing_models/MODE_fcstHRRR_obsMRMS_Hail_GRIB2.py]: Could you please remove the "n" from "nDiagnosticsUseCase"?

**Review by jprestop — CHANGES_REQUESTED** [review]
@lisagoodrich, Could you please take a look at this page and update it?
https://metplus.readthedocs.io/en/feature_963_quick_search/generated/model_applications/convection_allowing_models/EnsembleStat_fcstHRRR_fcstOnly_SurrogateSevere.html#sphx-glr-generated-model-applications-convection-allowing-models-ensemblestat-fcsthrrr-fcstonly-surrogatesevere-py

It looks like this one has not yet been updated.  I believe you'll find it at this relative path:  model_applications/convection_allowing_models/EnsembleStat_fcstHRRR_fcstOnly_SurrogateSevere

**Comment by lisagoodrich** [comment]
Thanks, Julie.  This wasn't in the original list that you gave me.
Hopefully this is the only one that has slipped through the cracks.  I have
made the updates and pushed this back up.  Please continue checking.
~Lisa

File: docs/use_cases/model_applications/convection_allowing_models/Point2Grid_obsLSR_ObsOnly_PracticallyPerfect.py [hunk 2]
```diff
@@ -141,19 +141,22 @@
...
-#   `ConvectionAllowingModelsAppUseCase <https://dtcenter.github.io/METplus/search.html?q=ConvectionAllowingModelsAppUseCase&check_keywords=yes&area=default>`_,
-#   `NCAROrgUseCase <https://dtcenter.github.io/METplus/search.html?q=NCAROrgUseCase&check_keywords=yes&area=default>`_,
-#   `ProbabilityGenerationUseCase <https://dtcenter.github.io/METplus/search.html?q=ProbabilityGenerationUseCase&check_keywords=yes&area=default>`_,
-#   `MaskingFeatureUseCase <https://dtcenter.github.io/METplus/search.html?q=MaskingFeatureUseCase&check_keywords=yes&area=default>`_ 
-#   `HMTOrgUseCase <https://dtcenter.github.io/METplus/search.html?q=HMTOrgUseCase&check_keywords=yes&area=default>`_ 
-#   `HWTOrgUseCase <https://dtcenter.github.io/METplus/search.html?q=HWTOrgUseCase&check_keywords=yes&area=default>`_ 
 #
-
+#   * ASCII2NC
+#   * Point2Gri
```
Comment (jprestop) [inline docs/use_cases/model_applications/convection_allowing_models/Point2Grid_obsLSR_ObsOnly_PracticallyPerfect.py]: It looks like there is a "d" missing in "Point2Grid" can you please add that back in?

Comment (jprestop) [reply docs/use_cases/model_applications/convection_allowing_models/Point2Grid_obsLSR_ObsOnly_PracticallyPerfect.py]: I fixed this up and made a few other changes to this file.

**Comment by jprestop** [comment]
@lisagoodrich You are correct - that wasn't in the original list that I gave you and snuck through.   I'll do a quick check for others.

**Review by jprestop — APPROVED** [review]
I pulled in changes from this branch and changes from develop.  I ran "find . -name "*.py" | xargs egrep -n -l "search.html" | more" and didn't find any other use_cases that needed to be fixed.  **We just need to wait for the checks to be completed before merging with the develop branch.**  I approve this request.

**Pull request merged**